addons/website_membership/controllers/main.py

```diff
@@ -46,41 +43,38 @@ def members(self, membership_id=None, country_name=None, country_id=0, page=1, *
         ]
         if membership_id and membership_id != 'free':
             membership_id = int(membership_id)
-            base_line_domain.append(('membership_id', '=', membership_id))
-            membership = product_obj.browse(cr, uid, membership_id, context=context)
-        else:
-            membership = None
+            base_line_domain += [
+                ('membership_id', '=', membership_id), ('date_to', '>=', today),
+                ('date_from', '<', today), ('state', '=', 'paid')]
+
         if post_name:
             base_line_domain += ['|', ('partner.name', 'ilike', post_name),
                                       ('partner.website_description', 'ilike', post_name)]
 
         # group by country, based on all customers (base domain)
         if membership_id != 'free':
-            membership_line_ids = membership_line_obj.search(cr, SUPERUSER_ID, base_line_domain, context=context)
-            country_domain = [('member_lines', 'in', membership_line_ids)]
+            membership_lines = MembershipLineSudo.search(base_line_domain)
+            country_domain = [('member_lines', 'in', membership_lines.ids)]
             if not membership_id:
                 country_domain = ['|', country_domain[0], ('membership_state', '=', 'free')]
         else:
-            membership_line_ids = []
             country_domain = [('membership_state', '=', 'free')]
         if post_name:
             country_domain += ['|', ('name', 'ilike', post_name),
-                                  ('website_description', 'ilike', post_name)]
-        countries = partner_obj.read_group(
-            cr, SUPERUSER_ID, country_domain + [("website_published", "=", True)], ["id", "country_id"],
-            groupby="country_id", orderby="country_id", context=request.context)
+                               ('website_description', 'ilike', post_name)]
+        countries = PartnerSudo.read_group(country_domain + [("website_published", "=", True)], ["id", "country_id"], groupby="country_id", orderby="country_id")
         countries_total = sum(country_dict['country_id_count'] for country_dict in countries)
```

Comment:
can you check sum aggregate operation required here explicitly ? read_group will not give you directly aggregated result (sum of country_id_count)  ?